kube_hunter/modules/hunting/kubelet.py

```diff
@@ -434,6 +452,453 @@ def is_kubesystem_pod(pod):
                     }
 
 
+""" Active Hunters """
+
+
+@handler.subscribe(AnonymousAuthEnabled)
+class FootholdViaSecureKubeletPort(ActiveHunter):
+    """Foothold Via Secure Kubelet Port
+    Attempts to demonstrate that a malicious actor can establish foothold into the cluster via a
+    container abusing the configuration of the kubelet's secure port: authentication-auth=false.
+    """
+
+    def __init__(self, event):
+        self.event = event
+        self.base_url = "https://{host}:10250/".format(host=self.event.host)
+
+    def get_request(self, url, verify=False):
+        try:
+            response_text = requests.get(url, verify).text.rstrip()
+
+            return response_text
+        except Exception as ex:
+            logging.debug("Exception: " + str(ex))
+            return "Exception: " + str(ex)
+
+    @staticmethod
+    def post_request(url, params, verify=False):
+        try:
+            response_text = requests.post(url, verify, params=params).text.rstrip()
```

Comment:
You want to use the self.event.session, this allows kube-hunter to use the already set authorization headers it got from previous discoveries such as when running from a pod.
So you might want to make this not a static method, and perhaps create a similar method in the "Malicous Intent" hunter, so the session variable will be available.

Reply:
Changed, including testing and the requirements file to support mocking sessions.